### PR TITLE
Fix Intel Compile errors with cxxopts and json.

### PR DESCRIPTION
**Pull request by mcfadden8**
Intel was finicky and required adjustments to these two files.

### CODE REVIEW

**Comment by mcfadden8** [comment]
@davidbeckingsale, @jdahm, this is ready for review.

File: .gitmodules [hunk 1]
```diff
@@ -6,4 +6,4 @@
 	url = https://github.com/LLNL/conduit.git
 [submodule "src/umpire/tpl/cxxopts"]
 	path = src/umpire/tpl/cxxopts
-	url = https://github.com/jarro2783/cxxopts.git
+  url = https://github.com/LLNL/cxxopts.git
```
Comment (jdahm) [inline .gitmodules]: Tabbing issue

File: src/umpire/tpl/json/json.hpp [hunk 2]
```diff
@@ -12610,6 +12610,14 @@ class serializer
         }
     }
 
+    template<typename NumberType, detail::enable_if_t<
```
Comment (jdahm) [inline src/umpire/tpl/json/json.hpp]: `std::enable_if_t` was added in C++14. Good, it looks like this is defined by the code on line 620 based on C++11 constructs.

File: .gitmodules (hunk 1 above)
Comment (mcfadden8) [reply .gitmodules]: Nice catch.  I believe I have addressed this now.

File: src/umpire/tpl/json/json.hpp (hunk 2 above)
Comment (mcfadden8) [reply src/umpire/tpl/json/json.hpp]: Is there something further I need to change here?  I'm not sure I follow the "Good" portion of the comment

Comment (jdahm) [reply src/umpire/tpl/json/json.hpp]: Sorry for the confusion. No, I was just commenting that it's nice that they defined `enable_if_t`!

**Comment by mcfadden8** [comment]
Thanks @jdahm,

@davidbeckingsale ? :-)

**Pull request merged**